tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,494 @@
+            await hass.async_block_till_done()
+
+        main_media_entity = hass.data['media_player'].get_entity(
+            MAIN_ENTITY_ID)
+
+        client_media_entity = hass.data['media_player'].get_entity(
+            CLIENT_ENTITY_ID)
+
+        # Set the client so it seems a recording is being watched.
+        client_media_entity.dtv.attributes = RECORDING
+        # Clients do not support turning on, setting it as client is on here.
+        client_media_entity.dtv._standby = False
+
+        main_media_entity.schedule_update_ha_state(True)
+        client_media_entity.schedule_update_ha_state(True)
+        await hass.async_block_till_done()
+
+        return (main_media_entity, client_media_entity)
+
+    return setup_test_platforms
+
+
+async def async_turn_on(hass, entity_id=None):
+    """Turn on specified media player or all."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_TURN_ON, data)
+
+
+async def async_turn_off(hass, entity_id=None):
+    """Turn off specified media player or all."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_TURN_OFF, data)
+
+
+async def async_media_pause(hass, entity_id=None):
+    """Send the media player the command for pause."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_PAUSE, data)
+
+
+async def async_media_play(hass, entity_id=None):
+    """Send the media player the command for play/pause."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_PLAY, data)
+
+
+async def async_media_stop(hass, entity_id=None):
+    """Send the media player the command for stop."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_STOP, data)
+
+
+async def async_media_next_track(hass, entity_id=None):
+    """Send the media player the command for next track."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_NEXT_TRACK, data)
+
+
+async def async_media_previous_track(hass, entity_id=None):
+    """Send the media player the command for prev track."""
+    data = {ATTR_ENTITY_ID: entity_id} if entity_id else {}
+    await hass.services.async_call(DOMAIN, SERVICE_MEDIA_PREVIOUS_TRACK, data)
+
+
+async def async_play_media(hass, media_type, media_id, entity_id=None,
+                           enqueue=None):
+    """Send the media player the command for playing media."""
+    data = {ATTR_MEDIA_CONTENT_TYPE: media_type,
+            ATTR_MEDIA_CONTENT_ID: media_id}
+
+    if entity_id:
+        data[ATTR_ENTITY_ID] = entity_id
+
+    if enqueue:
+        data[ATTR_MEDIA_ENQUEUE] = enqueue
+
+    await hass.services.async_call(DOMAIN, SERVICE_PLAY_MEDIA, data)
+
+
+class mockDIRECTVClass:
```

Comment:
`MockDIRECTV`

https://www.python.org/dev/peps/pep-0008/#class-names

Reply:
Based on that, should be: MockDirectvClass?

FYI, reason why I called it mockDIRECTVClass is cause the name of the original class is DIRECTV :-)